Update to work on Julia 1.9

blakejohnson opened the pull request:
I embarked on this before realizing that @matthewware had already done basically the same thing in #31. This doesn't include the updates he has to the APS file format, or the start he made to support bbndb.

blakejohnson committed:
Some updates to modern Julia syntax.

blakejohnson committed:
More modernization.

blakejohnson committed:
Finally able to create basic instructions (pulses).

blakejohnson committed:
More fixups

blakejohnson committed:
Some fixes for YAML wonkiness and making channels 34 actually work.

blakejohnson committed:
Julia 1.0+ package management. Fix test runner.

blakejohnson committed:
Remove ch34 stuff.

blakejohnson committed:
find -> findall and Void -> nothing

gribeill commented:
Wow, really bringing this one back from the grave, @blakejohnson! Thanks! 

QGL.jl and pyQGL have significantly diverged with how the channel library is defined and used (it's a sqlite database now...), so I'm not sure what the right move is for thr future of QGL.jl. @grahamrow any thoughts? This might be a good jumping off point for the lower level language/IR we've been talking about as we get ready for the APS4...

blakejohnson commented:
Understood. I think if you want to actually use this, you would need to update how it deals with channel libraries. For my purposes, I just wanted to create a speed benchmark, and so the old YAML format was sufficient.

blakejohnson committed:
Fix bugs in scheduling sequences with multiple logicals per physical.

NOTE: These changes haven't been tested with hardware. I merely found several examples where the code would not terminate because of how various indices were being tracked. So, these changes are sufficient to get the code to terminate.

blakejohnson committed:
Clean up some comments.
